src/main/java/ch/njol/skript/expressions/ExprHotbarSlot.java

```diff
@@ -20,70 +20,116 @@
 
 import org.bukkit.entity.Player;
 import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerItemHeldEvent;
 import org.bukkit.inventory.PlayerInventory;
 import org.eclipse.jdt.annotation.Nullable;
 
-import ch.njol.skript.classes.Changer;
+import ch.njol.skript.classes.Changer.ChangeMode;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
-import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.registrations.EventValues;
+import ch.njol.skript.util.Getter;
 import ch.njol.skript.util.slot.InventorySlot;
 import ch.njol.skript.util.slot.Slot;
+import ch.njol.util.Kleenean;
 
 @Name("Hotbar Slot")
-@Description({"The currently selected hotbar <a href='./classes.html#slot'>slot</a>. To retrieve its number use <a href='#ExprSlotIndex'>Slot Index</a> expression."})
-@Examples({"message \"%player's current hotbar slot%\"",
-            "set player's selected hotbar slot to slot 4 of player",
-			"send \"index of player's current hotbar slot = 1\" # second slot from the left"})
+@Description({
+	"The currently selected hotbar <a href='./classes.html#slot'>slot</a>.",
+	"To retrieve its number use <a href='#ExprSlotIndex'>Slot Index</a> expression.",
+	"Use future and past tense to grab the previous slot in an item change event, see example."
+})
+@Examples({
+	"message \"%player's current hotbar slot%\"",
+	"set player's selected hotbar slot to slot 4 of player",
+	"",
+	"send \"index of player's current hotbar slot = 1\" # second slot from the left",
```

Comment:
I know this was the original behaviour but why did we zero-index these? Most things in Skript start at 1.

Reply:
To signify that 1 is the second slot and not the first.

Reply:
I'm not sure how that relates to my question.